packages/synthetix-main/contracts/modules/SynthsModule.sol

```diff
@@ -0,0 +1,68 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@synthetixio/core-contracts/contracts/proxy/Beacon.sol";
+import "@synthetixio/core-contracts/contracts/ownership/OwnableMixin.sol";
+import "@synthetixio/core-contracts/contracts/proxy/BeaconProxy.sol";
+import "../storage/SynthsModuleStorage.sol";
+
+contract SynthsModule is OwnableMixin, SynthsModuleStorage {
+    error BeaconAlreadyDeployed();
+    error BeaconNotDeployed();
+    error ImplementationNotSet();
+    error SynthAlreadyDeployed();
+
+    event BeaconDeployed(address beacon);
+    event SynthDeployed(bytes32 synth, address synthProxyAddress);
+
+    function deployBeacon() external onlyOwner {
+        _deployBeacon();
```

Comment:
what if we move the code of _deployBeacon() here and remove the internal function

Reply:
Yeah +1
And if a function needs it and is not deployed, such function can just revert.

Reply:
So it is an internal function because it is reused. In the last discussion we had, we agreed that we want a lazy eval i.e. deploy everything not deployed when needed.

Reply:
I decided to remove the "lazy deployment" technique and have just reverts in case something is not set correctly. Now it has to be done via the deployment script. I also added the relevant tests.